.github/workflows/release.yml

```diff
@@ -0,0 +1,82 @@
+name: C/C++ Major/Minor Release CI
+
+on:
+  push:
+    tags:
+      - 'v*.*.*'  # Trigger the workflow on version tags like v1.0.0
+  release:
+    types: [created]
```

Comment:
@marshal-dteach I changed the trigger type to "main branch" and "push events" to prevent re-entrance caused by new tag pushing events. Is it possible that this scenario could occur again?

Reply:
@25077667 Can you tell me the exact way you want to put tags. I have tested it to run the with tags like v1.2.3 which will trigger the release.yml and without any tag it will trigger ci.yml and increment the version automatically based on latest release

Reply:
@marshal-dteach I have no preconceptions about the versioning rule. What are your thoughts on it?

Reply:
@25077667 current setup should be sufficient for tracking the versions as you can - 
on each commit test your build with the incremental patch build and have builds of previous commit in the release repository for comparison

and upgrade the minor / major version to track a specific change.

Reply:
@marshal-dteach Sorry for replying to you so late. I think your solution is okay for me. Thank you for your contribution.